src/services/initializer.ts

```diff
@@ -0,0 +1,117 @@
+import { CLIError } from "@oclif/errors"
+import * as cp from "node:child_process"
+import * as fs from "node:fs"
+import * as path from "node:path"
+import { BundleDescriptor } from "../models/bundle-descriptor"
+
+export interface InitializerOptions {
+  parentDirectory: string
+  name: string
+  version: string
+}
+
+/** Handles the scaffolding of a project bundle */
+export default class Initializer {
+  private readonly options: InitializerOptions
+
+  constructor(options: InitializerOptions) {
+    this.options = options
+  }
+
+  public async performScaffolding(): Promise<void> {
+    if (!/^[\w-]+$/.test(this.options.name)) {
+      throw new CLIError(`"${this.options.name}" is not a valid bundle name`)
+    }
+
+    this.createBundleDirectories()
+    this.createBundleDescriptor()
+    this.createDockerfile()
+    this.createGitignore()
+    this.createConfigJson()
+    this.initGitRepo()
+  }
+
+  private createBundleDirectories() {
+    const bundleDir = this.getBundleDirectory()
+
+    if (fs.existsSync(bundleDir)) {
+      throw new CLIError(`${bundleDir} already exists`)
+    }
+
+    try {
+      fs.accessSync(this.options.parentDirectory, fs.constants.W_OK)
+    } catch {
+      throw new CLIError(
+        `Directory ${this.options.parentDirectory} is not writable`
+      )
+    }
+
+    fs.mkdirSync(bundleDir)
+    fs.mkdirSync(this.getBundleFilePath(".ent"))
+    fs.mkdirSync(this.getBundleFilePath(".ent", "output"))
+    fs.mkdirSync(this.getBundleFilePath("microservices"))
+    fs.mkdirSync(this.getBundleFilePath("microfrontends"))
+    fs.mkdirSync(this.getBundleFilePath("epc"))
+  }
+
+  private createBundleDescriptor() {
+    const bundleDescriptorFile = this.getBundleFilePath("bundle.json")
+
+    const bundleDescriptor: BundleDescriptor = {
+      name: this.options.name,
+      version: this.options.version,
+      microservices: [],
+      microfrontends: []
+    }
+
+    fs.writeFileSync(
+      bundleDescriptorFile,
+      JSON.stringify(bundleDescriptor, null, 4)
+    )
+  }
+
+  private createDockerfile() {
+    this.createFileFromTemplate(
+      this.getBundleFilePath("Dockerfile"),
+      "Dockerfile-template"
+    )
+  }
+
+  private createGitignore() {
+    this.createFileFromTemplate(
+      this.getBundleFilePath(".gitignore"),
+      "gitignore-template"
+    )
+  }
+
+  private createConfigJson() {
+    // TODO: ENG-3623
+  }
+
+  private createFileFromTemplate(filePath: string, templateFileName: string) {
+    const templateFileContent = fs.readFileSync(
+      path.resolve(__dirname, "..", "resources", templateFileName)
```

Comment:
I wanted to load the default `Dockerfile` and `.gitignore` from template files located inside sources folder, to avoid hardcoding their content directly inside the class, so I used `__dirname` here.
I've seen a `Do not use "__dirname"` warning and I've read that the preferred method should be using `import.meta.url`. However that construct is not available for commonjs modules (that we specified in the `tsconfig.json`). Can one of our JS experts recommend the best approach for our use case?

Reply:
I think for our current context (Node.js 14) we can ignore this warning for this reason:

`import.meta` is introduced by ESM for retrieving information about the current module (is the future way) and in the Node.js context will be a replacement for `__dirname` and `__filename`, there is some problem for our case:

Node.js 14 ESM supports is experimental (https://nodejs.org/dist./v14.10.0/docs/api/esm.html#esm_modules_ecmascript_modules) and not predictable.
`import.meta `is interpreted at runtime, in our case the ESM is purely static because we are going to transpile import to common.js modules so when the program run we effectively don't use ESM.

My advice here is to continue to use `__dirname` and `__filename`, when we schedule future updates of node.js for Entando we can think about changing the runtime module system to be compliant with the future.

Reply:
How can we ignore this in the linter?

Reply:
add to `.eslintrc` this rule:

```json
"rules": {
   "unicorn/prefer-module": "off"
}
```

Reply:
Thanks!